src/main/c/com/micromouse/robot/robot.c

```diff
@@ -182,4 +182,60 @@ void RobotDestructor(Robot* winslow) {
 
 	// Free up the memory for the Robot
 	free(winslow);
-}
\ No newline at end of file
+}
+
+// Robot Location -> [Linked-list of [Moves]]
+// Returns a linked list of moves for the robot to execute, starts by 
+// searching the maze for the path to the goal, and returns the path as a 
+// Linked-list of Moves.
+Cell* SolveMaze(Robot* winslow, Location *goal) {
+  // Create path to return and load starting location into path
+  Location* curr_loc = winslow->location_;
+
+  // Mark curr_loc as visited
+  int curr_x = curr_loc->x;
+  int curr_y = curr_loc->y;
+  List* queue = InitializeList(winslow->maze_[curr_x][curr_y]);
+  Cell* curr_cell = winslow->maze_[curr_x][curr_y];
+  curr_cell->visited = 1;
+
+  int is_queue_empty = empty(queue);
+  while(!is_queue_empty) {
+    // TODO: Dequeue curr_cell from queue.
+    curr_cell = Front(&queue);
+    if(curr_cell->location == goal) {
+      return winslow->maze_[goal->x][goal->y];
+    }
+    curr_loc = curr_cell->location;
+    int x_ = curr_loc->x;
+    int y_ = curr_loc->y;
+    if(curr_cell->east) {
+      queue = VisitNeighbor(curr_cell, x_ + 1, y_, winslow, queue);
+    }
+    if(curr_cell->west) {
+      queue = VisitNeighbor(curr_cell, x_ - 1, y_, winslow, queue);
+    }
+    if(curr_cell->north) {
+      queue = VisitNeighbor(curr_cell, x_, y_ + 1, winslow, queue);
+    }
+    if(curr_cell->south) {
+      queue = VisitNeighbor(curr_cell, x_, y_ - 1, winslow, queue);
+    }
+    is_queue_empty = empty(queue);
+  }
+  return winslow->maze_[0][0]; // Did not find goal cell, return fail.
```

Comment:
it might be better to just return NULL here. Because how do you know that [0][0] will always be a fail. What if the goal happens to be [0][0] then how do you distinguish it from a failure